snakes/go/pathy-snake/pathing.go

```diff
@@ -125,24 +125,48 @@ func getTargetCell(state GameState, grid *Grid) *Cell {
 		targetCell = chooseNearestFood(grid, state)
 	}
 
-	// // If targetCell is nil, then move in a direction that is further away from the larger snakes.
-	// if targetCell == nil {
-	// 	targetCell = moveAwayFromLargerSnakes(grid, state)
-	// }
-
-	// If we still don't have a target cell, then just pick a random walkable cell.
-	if targetCell == nil {
-		targetCell = chooseRandomWalkableTargetCell(grid, state)
-	}
-
-	// If we still don't have a target cell, then just pick a random cell.
-	if targetCell == nil {
-		targetCell = chooseRandomTargetCell(grid)
-	}
+	walkableCells := grid.CellsByWalkable(true)
+	targetCell = chooseTargetCell(state, grid, walkableCells)
 
 	return targetCell
 }
 
+// Check if the cell is in a hazard.
+func isCellInHazard(state GameState, cell *Cell) bool {
+    for _, hazard := range state.Board.Hazards {
+        if hazard.X == cell.X && hazard.Y == cell.Y {
+            return true
+        }
+    }
+    return false
+}
+
+// Flood fill function
+func floodFill(state GameState, grid *Grid, cell *Cell, visited map[*Cell]bool, depth, maxDepth int) int {
+    if _, ok := visited[cell]; ok || depth > maxDepth {
+        return 0
+    }
+
+    visited[cell] = true
+
+    adjacentCells := []*Cell{
+        grid.Get(cell.X, (cell.Y+1)%state.Board.Height),
+        grid.Get(cell.X, (cell.Y-1+state.Board.Height)%state.Board.Height),
+        grid.Get((cell.X+1)%state.Board.Width, cell.Y),
+        grid.Get((cell.X-1+state.Board.Width)%state.Board.Width, cell.Y),
+    }
```

Comment:
We might need to account for when Pathy is on the edge here based on game modes. 

In standard games for example there is now X-1 or Y-1 when at 0,0